src/contracts/misc/GhoStewardV2.sol

```diff
@@ -0,0 +1,286 @@
+    require(owner != address(0), 'INVALID_OWNER');
+    require(addressesProvider != address(0), 'INVALID_ADDRESSES_PROVIDER');
+    require(ghoToken != address(0), 'INVALID_GHO_TOKEN');
+    require(riskCouncil != address(0), 'INVALID_RISK_COUNCIL');
+
+    POOL_ADDRESSES_PROVIDER = addressesProvider;
+    GHO_TOKEN = ghoToken;
+    RISK_COUNCIL = riskCouncil;
+
+    _transferOwnership(owner);
+  }
+
+  /// @inheritdoc IGhoStewardV2
+  function updateFacilitatorBucketCapacity(
+    address facilitator,
+    uint128 newBucketCapacity
+  ) external onlyRiskCouncil notTimelocked(_facilitatorsBucketCapacityTimelocks[facilitator]) {
+    require(_controlledFacilitatorsByAddress[facilitator], 'FACILITATOR_NOT_CONTROLLED');
+    (uint256 currentBucketCapacity, ) = IGhoToken(GHO_TOKEN).getFacilitatorBucket(facilitator);
+    require(
+      _isIncreaseLowerThanMax(currentBucketCapacity, newBucketCapacity, currentBucketCapacity),
+      'INVALID_BUCKET_CAPACITY_UPDATE'
+    );
+
+    _facilitatorsBucketCapacityTimelocks[facilitator] = uint40(block.timestamp);
+
+    IGhoToken(GHO_TOKEN).setFacilitatorBucketCapacity(facilitator, newBucketCapacity);
+  }
+
+  /// @inheritdoc IGhoStewardV2
+  function updateGhoBorrowCap(
+    uint256 newBorrowCap
+  ) external onlyRiskCouncil notTimelocked(_ghoTimelocks.ghoBorrowCapLastUpdate) {
+    DataTypes.ReserveConfigurationMap memory configuration = IPool(
+      IPoolAddressesProvider(POOL_ADDRESSES_PROVIDER).getPool()
+    ).getConfiguration(GHO_TOKEN);
+    uint256 currentBorrowCap = configuration.getBorrowCap();
+    require(
+      _isIncreaseLowerThanMax(currentBorrowCap, newBorrowCap, currentBorrowCap),
+      'INVALID_BORROW_CAP_UPDATE'
+    );
+
+    _ghoTimelocks.ghoBorrowCapLastUpdate = uint40(block.timestamp);
+
+    IPoolConfigurator(IPoolAddressesProvider(POOL_ADDRESSES_PROVIDER).getPoolConfigurator())
+      .setBorrowCap(GHO_TOKEN, newBorrowCap);
+  }
+
+  /// @inheritdoc IGhoStewardV2
+  function updateGhoBorrowRate(
+    uint256 newBorrowRate
+  ) external onlyRiskCouncil notTimelocked(_ghoTimelocks.ghoBorrowRateLastUpdate) {
+    DataTypes.ReserveData memory ghoReserveData = IPool(
+      IPoolAddressesProvider(POOL_ADDRESSES_PROVIDER).getPool()
+    ).getReserveData(GHO_TOKEN);
+    require(
+      ghoReserveData.interestRateStrategyAddress != address(0),
+      'GHO_INTEREST_RATE_STRATEGY_NOT_FOUND'
+    );
+
+    uint256 currentBorrowRate = GhoInterestRateStrategy(ghoReserveData.interestRateStrategyAddress)
+      .getBaseVariableBorrowRate();
+    require(newBorrowRate <= GHO_BORROW_RATE_MAX, 'BORROW_RATE_HIGHER_THAN_MAX');
+    require(
+      _isIncreaseLowerThanMax(currentBorrowRate, newBorrowRate, GHO_BORROW_RATE_CHANGE_MAX),
+      'INVALID_BORROW_RATE_UPDATE'
+    );
+
+    address cachedStrategyAddress = _ghoBorrowRateStrategiesByRate[newBorrowRate];
+    if (cachedStrategyAddress == address(0)) {
+      GhoInterestRateStrategy newRateStrategy = new GhoInterestRateStrategy(
```

Comment:
as i understand there's nothing really "GHO" specific about this ir. It's just a constant ir. Wouldn't it make more sense to externalize the ir registry/factory in a similar pattern as https://github.com/bgd-labs/aave-helpers/blob/master/src/v3-config-engine/V3RateStrategyFactory.sol ?

Reply:
The `GhoInterestRateStrategy` offers a constant borrow rate. While the `DefaultReserveInterestRateStrategy` could also provide a constant rate, it adds unnecessary complexity in a less efficient way. We also prefer to keep this implementation as it is, to speed up the its deployment.

Reply:
I've updated the implementation with a new contract [FixedRateStrategyFactory](https://github.com/aave/gho-core/blob/feat/gho-steward-v2/src/contracts/misc/FixedRateStrategyFactory.sol) which is a simplified version of the one you shared, using [GhoInterestRateStrategy](https://github.com/aave/gho-core/blob/feat/gho-steward-v2/src/contracts/facilitators/aave/interestStrategy/GhoInterestRateStrategy.sol). I've used `Fixed` instead of `Constant` because i think it't more consistent with how we name in this repo.
Let me know what you think about this.